### ISSUE TITLE
nis_soss spec2 error for "zero-size array to reduction operation maximum which has no identity"

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#8166]
_Issue [JP-3499](https://jira.stsci.edu/browse/JP-3499) was created on JIRA by [Hien Tran](https://jira.stsci.edu/secure/ViewProfile.jspa?name=htran):_

ops saw a level-2b (spec2) failure during reprocessing of nis_soss dataset [jw01201-o008_20231219t214518_tso-spec2_00002](https://pljwdmsweb.stsci.edu/owlgui/logs/directory_view?directory_path=/ifs/archive/ops/jwst/info/owl/logs/owlmgr_jw01201-o008_20231219t214518_tso-spec2_00002_1703023726.653522&fileset_name=jw01201-o008_20231219t214518_tso-spec2_00002&dag_name=REPRO_LEVEL_2B) ((jw01201008001_04101_00001-{*}seg003{*}_nis)) with dms build 10.0 in the extract-1d step. this appears to be a regression, as this particular fileset has made 2b successfully many times before during repro in previous builds. the traceback, extracted below, can be found at Directory Path: /ifs/archive/ops/jwst/info/owl/logs/owlmgr_jw01201-o008_20231219t214518_tso-spec2_00002_1703023726.653522
```java
2023-12-19 22:09:25,174 - stpipe.Spec2Pipeline.pixel_replace - INFO - Step skipped.
2023-12-19 22:09:25,177 - stpipe.Spec2Pipeline.pixel_replace - INFO - Step pixel_replace done
2023-12-19 22:09:27,994 - stpipe.Spec2Pipeline.extract_1d - INFO - Step extract_1d running with args (<CubeModel(186, 256, 2048) from jw01201008001_04101_00001-seg003_nis_calints.fits>,).
2023-12-19 22:09:27,995 - stpipe.Spec2Pipeline.extract_1d - INFO - Step extract_1d parameters are: {'pre_hooks': [], 'post_hooks': [], 'output_file': None, 'output_dir': None, 'output_ext': '.fits', 'output_use_model': False, 'output_use_index': True, 'save_results': False, 'skip': False, 'suffix': 'x1dints', 'search_output_file': True, 'input_dir': '/ifs/archive/ops/jwst/info/owlmgr/paths/sdp/asn_creation/cal/level2', 'smoothing_length': None, 'bkg_fit': None, 'bkg_order': None, 'bkg_sigma_clip': 3.0, 'log_increment': 50, 'subtract_background': None, 'use_source_posn': None, 'center_xy': None, 'apply_apcorr': True, 'ifu_autocen': False, 'ifu_rfcorr': False, 'ifu_set_srctype': None, 'ifu_rscale': None, 'soss_atoca': True, 'soss_threshold': 0.01, 'soss_n_os': 2, 'soss_wave_grid_in': None, 'soss_wave_grid_out': None, 'soss_estimate': None, 'soss_rtol': 0.0001, 'soss_max_grid_size': 20000, 'soss_transform': None, 'soss_tikfac': None, 'soss_width': 40.0, 'soss_bad_pix': 'masking', 'soss_modelname': None}
2023-12-19 22:09:28,006 - stpipe.Spec2Pipeline.extract_1d - INFO - Input is a NIRISS SOSS observation, the specialized SOSS extraction (ATOCA) will be used.
2023-12-19 22:09:28,007 - stpipe.Spec2Pipeline.extract_1d - INFO - Exposure is through the GR700XD + CLEAR (science).
2023-12-19 22:09:28,007 - stpipe.Spec2Pipeline.extract_1d - INFO - Exposure is in the SUBSTRIP256 subarray.
2023-12-19 22:09:28,007 - stpipe.Spec2Pipeline.extract_1d - INFO - Traces 1 and 2 will be modelled and decontaminated before extraction.
2023-12-19 22:09:28,019 - CRDS - DEBUG - ========================================================================================================================
2023-12-19 22:09:28,019 - CRDS - DEBUG - getreferences() CRDS version: 11.17.6, b11.4.0, 64d96076d89b32a5687a6b77bb910ab93b3a99b3
2023-12-19 22:09:28,019 - CRDS - DEBUG - getreferences() server: https://jwst-serverless-mode.stsci.edu
2023-12-19 22:09:28,019 - CRDS - DEBUG - getreferences() observatory: jwst
2023-12-19 22:09:28,019 - CRDS - DEBUG - getreferences() reftypes: ('spectrace',)
2023-12-19 22:09:28,019 - CRDS - DEBUG - getreferences() context: None
2023-12-19 22:09:28,019 - CRDS - DEBUG - getreferences() ignore_cache: False
2023-12-19 22:09:28,019 - CRDS - DEBUG - CRDS_PATH = /ifs/archive/ops/jwst/ref/tmp_crds/crds/cache
2023-12-19 22:09:28,019 - CRDS - DEBUG - CRDS_SERVER_URL = https://jwst-serverless-mode.stsci.edu
2023-12-19 22:09:28,019 - CRDS - DEBUG - Final effective context is 'jwst_1177.pmap'
2023-12-19 22:09:28,020 - CRDS - DEBUG - Computing best references locally.
2023-12-19 22:09:28,020 - CRDS - DEBUG - Bestrefs header:
{'META.EXPOSURE.READPATT [READPATT]': 'NISRAPID',
'META.EXPOSURE.TYPE [EXP_TYPE]': 'NIS_SOSS',
'META.INSTRUMENT.DETECTOR [DETECTOR]': 'NIS',
'META.INSTRUMENT.FILTER [FILTER]': 'CLEAR',
'META.INSTRUMENT.LAMP_STATE [LAMP]': 'NONE',
'META.INSTRUMENT.NAME [INSTRUME]': 'NIRISS',
'META.INSTRUMENT.PUPIL [PUPIL]': 'GR700XD',
'META.OBSERVATION.DATE [DATE-OBS]': '2023-06-11',
'META.OBSERVATION.TIME [TIME-OBS]': '20:12:43.261',
'META.SUBARRAY.NAME [SUBARRAY]': 'SUBSTRIP256',
'META.VISIT.CROWDED_FIELD [CROWDFLD]': 'F',
'META.VISIT.TSOVISIT [TSOVISIT]': 'T',
'META.VISIT.TYPE [VISITYPE]': 'PRIME_TARGETED_FIXED',
'REFTYPE': 'UNDEFINED'}


...


2023-12-19 22:09:28,076 - CRDS - DEBUG - Reference type 'speckernel' defined as 'jwst_niriss_speckernel_0004.fits'
2023-12-19 22:09:28,310 - stpipe.Spec2Pipeline.extract_1d - INFO - Wavelength grid was not specified. Setting `wave_grid` to None.
2023-12-19 22:09:28,373 - stpipe.Spec2Pipeline.extract_1d - INFO - Input is a CubeModel containing 186 integrations.
2023-12-19 22:09:28,373 - stpipe.Spec2Pipeline.extract_1d - INFO - Processing integration 1 of 186.
2023-12-19 22:09:28,380 - stpipe.Spec2Pipeline.extract_1d - INFO - Skip background subtraction.
2023-12-19 22:09:28,380 - stpipe.Spec2Pipeline.extract_1d - INFO - Solving for the transformation parameters.
2023-12-19 22:09:28,381 - stpipe.Spec2Pipeline.extract_1d - INFO - Parameters used for fit: theta, x-offset, y-offset
2023-12-19 22:09:28,381 - stpipe.Spec2Pipeline.extract_1d - INFO - Fixed parameters:
2023-12-19 22:09:28,610 - stpipe.Spec2Pipeline.extract_1d - INFO - Measuring trace position for orders 1 and 2.
2023-12-19 22:09:28,851 - stpipe.Spec2Pipeline.extract_1d - INFO - Measured to Reference trace position transform: theta=0.07949012186071586, x-offset=-1.3197735196129754, y-offset=-0.7968755559305017
2023-12-19 22:09:30,886 - stpipe.Spec2Pipeline.extract_1d - INFO - wave_grid not given: generating grid based on rtol=0.0001
2023-12-19 22:12:29,225 - stpipe.Spec2Pipeline.extract_1d - INFO - Solving for the optimal Tikhonov factor.
2023-12-19 22:12:29,229 - stpipe.Spec2Pipeline.extract_1d - INFO - First guess of tikhonov factor: 6.818426213134753e-17
2023-12-19 22:12:52,733 - stpipe.Spec2Pipeline.extract_1d - INFO - Testing factors...
2023-12-19 22:12:57,951 - stpipe.Spec2Pipeline.extract_1d - INFO - 0/10
2023-12-19 22:13:03,134 - stpipe.Spec2Pipeline.extract_1d - INFO - 1/10
2023-12-19 22:13:08,296 - stpipe.Spec2Pipeline.extract_1d - INFO - 2/10
2023-12-19 22:13:13,446 - stpipe.Spec2Pipeline.extract_1d - INFO - 3/10
2023-12-19 22:13:18,638 - stpipe.Spec2Pipeline.extract_1d - INFO - 4/10
2023-12-19 22:13:23,835 - stpipe.Spec2Pipeline.extract_1d - INFO - 5/10
2023-12-19 22:13:29,011 - stpipe.Spec2Pipeline.extract_1d - INFO - 6/10
2023-12-19 22:13:34,144 - stpipe.Spec2Pipeline.extract_1d - INFO - 7/10
2023-12-19 22:13:39,181 - stpipe.Spec2Pipeline.extract_1d - INFO - 8/10
2023-12-19 22:13:44,003 - stpipe.Spec2Pipeline.extract_1d - INFO - 9/10
2023-12-19 22:13:44,003 - stpipe.Spec2Pipeline.extract_1d - INFO - 10/10
2023-12-19 22:14:06,976 - stpipe.Spec2Pipeline.extract_1d - INFO - Testing factors...
2023-12-19 22:14:12,293 - stpipe.Spec2Pipeline.extract_1d - INFO - 0/20
2023-12-19 22:14:17,655 - stpipe.Spec2Pipeline.extract_1d - INFO - 1/20


...


2023-12-20 02:27:31,792 - stpipe.Spec2Pipeline.extract_1d - INFO - 20/20
2023-12-20 02:27:32,220 - stpipe.Spec2Pipeline.extract_1d - INFO - Performing the box extraction.
2023-12-20 02:27:32,220 - stpipe.Spec2Pipeline.extract_1d - INFO - Bad pixels in Order 1 will be masked.
2023-12-20 02:27:32,251 - stpipe.Spec2Pipeline.extract_1d - INFO - Bad pixels in Order 2 will be masked.
2023-12-20 02:27:32,280 - stpipe.Spec2Pipeline.extract_1d - INFO - Bad pixels in Order 3 will be masked.
2023-12-20 02:27:32,386 - stpipe.Spec2Pipeline.extract_1d - INFO - Processing integration 72 of 186.
2023-12-20 02:27:32,390 - stpipe.Spec2Pipeline.extract_1d - INFO - Skip background subtraction.
2023-12-20 02:27:32,391 - stpipe.Spec2Pipeline.extract_1d - INFO - Measured to Reference trace position transform: theta=0.07949012186071586, x-offset=-1.3197735196129754, y-offset=-0.7968755559305017
2023-12-20 02:27:34,094 - stpipe.Spec2Pipeline.extract_1d - INFO - Using previously computed or user specified wavelength grid.
2023-12-20 02:30:07,479 - stpipe.Spec2Pipeline.extract_1d - INFO - Using a Tikhonov factor of 4.0077383338590684e-17
2023-12-20 02:30:13,434 - stpipe.Spec2Pipeline.extract_1d - INFO - Optimal solution has a log-likelihood of -751660.4863385509
2023-12-20 02:30:17,128 - stpipe.Spec2Pipeline.extract_1d - INFO - Generate model for blue-most part of order 2
Traceback (most recent call last):
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/pipeline/calwebb_spec2.py", line 122, in process
result = self.process_exposure_product(
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/pipeline/calwebb_spec2.py", line 322, in process_exposure_product
x1d = self.extract_1d(x1d)
^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/stpipe/step.py", line 478, in run
step_result = self.process(*args)
^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/extract_1d/extract_1d_step.py", line 456, in process
result, ref_outputs, atoca_outputs = soss_extract.run_extract1d(
^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/extract_1d/soss_extract/soss_extract.py", line 1183, in run_extract1d
result = model_image(scidata_bkg, scierr, scimask, refmask, ref_files, box_weights, subarray, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/extract_1d/soss_extract/soss_extract.py", line 692, in model_image
model, spec_ord = model_single_order(scidata_bkg, scierr, ref_file_order,
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/extract_1d/soss_extract/soss_extract.py", line 817, in model_single_order
engine = ExtractionEngine(*ref_file_args,
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/extract_1d/soss_extract/atoca.py", line 1551, in __init__
super().__init__(wave_map, trace_profile, *args, **kwargs)
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/extract_1d/soss_extract/atoca.py", line 205, in __init__
self.weights, self.weights_k_idx = self.compute_weights()
^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/extract_1d/soss_extract/atoca.py", line 573, in compute_weights
weights_n, k_idx_n = self.get_w(i_order) # Compute weights
^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/extract_1d/soss_extract/atoca.py", line 1699, in get_w
k_n, bad = atoca_utils.arange_2d(k_first, k_last + 1, dtype=int)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/extract_1d/soss_extract/atoca_utils.py", line 70, in arange_2d
ncols = np.amax(lengths)
^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/numpy/core/fromnumeric.py", line 2827, in amax
return _wrapreduction(a, np.maximum, 'max', axis, None, out,
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/numpy/core/fromnumeric.py", line 88, in _wrapreduction
return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: zero-size array to reduction operation maximum which has no identity
Traceback (most recent call last):
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/bin/strun", line 26, in <module>
step = Step.from_cmdline(sys.argv[1:])
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/stpipe/step.py", line 186, in from_cmdline
return cmdline.step_from_cmdline(args)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/stpipe/cmdline.py", line 386, in step_from_cmdline
step.run(*positional)
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/stpipe/step.py", line 478, in run
step_result = self.process(*args)
^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/pipeline/calwebb_spec2.py", line 140, in process
raise RuntimeError('\n'.join(failures))
RuntimeError: Traceback (most recent call last):
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/pipeline/calwebb_spec2.py", line 122, in process
result = self.process_exposure_product(
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/pipeline/calwebb_spec2.py", line 322, in process_exposure_product
x1d = self.extract_1d(x1d)
^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/stpipe/step.py", line 478, in run
step_result = self.process(*args)
^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/extract_1d/extract_1d_step.py", line 456, in process
result, ref_outputs, atoca_outputs = soss_extract.run_extract1d(
^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/extract_1d/soss_extract/soss_extract.py", line 1183, in run_extract1d
result = model_image(scidata_bkg, scierr, scimask, refmask, ref_files, box_weights, subarray, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/extract_1d/soss_extract/soss_extract.py", line 692, in model_image
model, spec_ord = model_single_order(scidata_bkg, scierr, ref_file_order,
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/extract_1d/soss_extract/soss_extract.py", line 817, in model_single_order
engine = ExtractionEngine(*ref_file_args,
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/extract_1d/soss_extract/atoca.py", line 1551, in __init__
super().__init__(wave_map, trace_profile, *args, **kwargs)
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/extract_1d/soss_extract/atoca.py", line 205, in __init__
self.weights, self.weights_k_idx = self.compute_weights()
^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/extract_1d/soss_extract/atoca.py", line 573, in compute_weights
weights_n, k_idx_n = self.get_w(i_order) # Compute weights
^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/extract_1d/soss_extract/atoca.py", line 1699, in get_w
k_n, bad = atoca_utils.arange_2d(k_first, k_last + 1, dtype=int)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/extract_1d/soss_extract/atoca_utils.py", line 70, in arange_2d
ncols = np.amax(lengths)
^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/numpy/core/fromnumeric.py", line 2827, in amax
return _wrapreduction(a, np.maximum, 'max', axis, None, out,
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/numpy/core/fromnumeric.py", line 88, in _wrapreduction
return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: zero-size array to reduction operation maximum which has no identity
2023354023019 INFO src=ssb_calibration_wrapper._strun_analyze_and_log_failure fsn=jw01201-o008_20231219t214518_tso-spec2_00002_asn msg="strun /dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.12.5.20231019-py3.11/lib/python3.11/site-packages/jwst/pipeline/calwebb_tso-spec2.cfg FAILED (exit=1) on jw01201-o008_20231219t214518_tso-spec2_00002_asn.json."
2023354023019 ERROR src=ssb_calibration_wrapper.calibrate._strun._strun_analyze_and_log_failure fsn=jw01201-o008_20231219t214518_tso-spec2_00002_asn msg="strun exit status=1" ```

**Comment by stscijgbot-jp** [comment]
_Comment by [Ned Molter](https://jira.stsci.edu/secure/ViewProfile.jspa?name=emolter) on [JIRA](https://jira.stsci.edu/browse/JP-3499?focusedCommentId=682240#comment-682240):_

Doing some digging, this is happening because the data quality flags (sci_mask) and the allowed wavelengths (mask_wave) do not overlap for this dataset.  See the attached figure.  The wavelength mask looks correct to me for order 2, so no problems there - looks to me like this integration just ended up with a lot of bad data. 

 

The question is, how should this case be handled?  For example, should I log the error, then simply set the flux values associated with order 2 to NaN or a dummy value?  Should the integration be skipped entirely?

 

The sci_mask contains a very small number of pixels that are considered valid beyond the very obvious wavelength cutoff; there are exactly 6 such pixels beyond index 1000. You can't see it in the figure, but I promise these are there.  This leads to exactly one pixel that is valid across all three masks.  It doesn't currently look like this should make a difference for squashing this bug, but it might be worth looking into to avoid future problems.  It does make it a bit more challenging to catch the error using a simple "no overlapping pixels between sci_mask and wave_mask," rule, though.

**Comment by stscijgbot-jp** [comment]
_Comment by [Ned Molter](https://jira.stsci.edu/secure/ViewProfile.jspa?name=emolter) on [JIRA](https://jira.stsci.edu/browse/JP-3499?focusedCommentId=682550#comment-682550):_

I dug a bit more into the reason why there are so many flags.  The problem is already apparent in the uncal file - see the attached figure (uncal_problem.png), in which all five integrations are plotted on the same brightness scale.  It looks like the detector1pipeline does a good job catching this, I assume in the jump detection step.  I don't know why it is happening at the instrument level - maybe someone else knows.  In any case, I'm going to go ahead and just catch a "too few good pixels to extract order 2" exception in Extract1d and set order 2 to be all NaNs.

**Comment by stscijgbot-jp** [comment]
_Comment by [Ned Molter](https://jira.stsci.edu/secure/ViewProfile.jspa?name=emolter) on [JIRA](https://jira.stsci.edu/browse/JP-3499?focusedCommentId=683026#comment-683026):_

RE the small number of pixels that remained unflagged in the bad region, this issue is already present at the end of the detector1pipeline; see the image named dq_rateints_int72.png, which shows a zoom-in on the uncal data and the data quality flags. There's a single unflagged pixel, which appears white, at index (1188, 104), which is a hot pixel in the uncal data.  So perhaps this requires an update to the bad pixel map in CRDS?  Unless people have other opinions, I'm planning to drop this now, because the workaround is easy enough.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-3499?focusedCommentId=683859#comment-683859):_

Fixed by #8265

**Comment by stscijgbot-jp** [comment]
_Comment by [Joseph Filippazzo](https://jira.stsci.edu/secure/ViewProfile.jspa?name=jfilippazzo) on [JIRA](https://jira.stsci.edu/browse/JP-3499?focusedCommentId=696656#comment-696656):_

I installed a fresh python 3.10 env and installed jwst-1.14.0 and tested the DETECTOR1 and SPEC2 reduction pipelines with the dataset `jw01201008001_04101_00001-seg001_nis_uncal.fits` and everything ran through with no problems! I can confirm this appears fixed. The notebook I made with all the steps is located at `/ifs/jwst/wit/niriss/pipeline_testing/build_10.2/JP-3499/JP-34{+}99{+} Testing.ipynb`.
